src/local_package.ml

```diff
@@ -10,8 +12,22 @@ type t =
   ; pkg : Package.t
   ; libs : Lib.Set.t
   ; virtual_lib : Lib.t option Lazy.t
+  ; id : Id.t
```

Comment:
I think we should not have the id and remove the hash function and use physical equality instead.
The motivation is to no have multiple objects around with different contents that compare equal unnecessarily.
I pushed the commit that makes that change. Feel free to revert that if you disagree or if I'm missing the point.